File: APP/TaQuot/src/models/Application/Application.ts
export default class ApplicationDTO {
  id = "";
  libelle = "";
  commanditaireId = "";
}

export class Application extends ApplicationDTO {
  public constructor(dto: ApplicationDTO | null = null) {
    super();

    if (dto?.id) {
      this.id = dto.id;
    }

    if (dto?.libelle) {
      this.libelle = dto.libelle;
    }

    if (dto?.commanditaireId) {
      this.commanditaireId = dto.commanditaireId;
    }
  }
}

File: API/AcQua TaQuot/Entities/Models/Application.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities.Models
{
    public class Application
    {
        [Column("ApplicationId")]
        public Guid Id { get; set; }
        [Required()]
        public string Libelle { get; set; }

        [ForeignKey(nameof(Statut))]
        public Guid? StatutId { get; set; }


        [ForeignKey(nameof(Commanditaire))]
        public Guid CommanditaireId { get; set; }
        public Commanditaire? Commanditaire { get; set; }
        public ICollection<Module>? Modules { get; set; }
    }
}
